src/zenml/log_stores/artifact/artifact_log_store.py

```diff
@@ -0,0 +1,329 @@
+
+    Args:
+        artifact_store: The artifact store on which the artifact will be stored.
+        log_id: The ID of the logs entity
+
+    Returns:
+        The URI of the log storage (file or folder).
+    """
+    logs_base_uri = os.path.join(artifact_store.path, "logs")
+
+    if artifact_store.config.IS_IMMUTABLE_FILESYSTEM:
+        logs_uri = os.path.join(logs_base_uri, str(log_id))
+    else:
+        logs_uri = os.path.join(logs_base_uri, f"{log_id}{LOGS_EXTENSION}")
+
+    return sanitize_remote_path(logs_uri)
+
+
+def remove_ansi_escape_codes(text: str) -> str:
+    """Auxiliary function to remove ANSI escape codes from a given string.
+
+    Args:
+        text: the input string
+
+    Returns:
+        the version of the input string where the escape codes are removed.
+    """
+    return ansi_escape.sub("", text)
+
+
+def fetch_log_records(
+    zen_store: "BaseZenStore",
+    artifact_store_id: Union[str, UUID],
+    logs_uri: str,
+) -> List[LogEntry]:
+    """Fetches log entries.
+
+    Args:
+        zen_store: The store in which the artifact is stored.
+        artifact_store_id: The ID of the artifact store.
+        logs_uri: The URI of the artifact (file or directory).
+
+    Returns:
+        List of log entries.
+    """
+    log_entries = []
+
+    for line in _stream_logs_line_by_line(
+        zen_store, artifact_store_id, logs_uri
+    ):
+        if log_entry := parse_log_entry(line):
+            log_entries.append(log_entry)
+
+        if len(log_entries) >= MAX_ENTRIES_PER_REQUEST:
+            break
+
+    return log_entries
+
+
+def _stream_logs_line_by_line(
+    zen_store: "BaseZenStore",
+    artifact_store_id: Union[str, UUID],
+    logs_uri: str,
+) -> Iterator[str]:
+    """Stream logs line by line without loading the entire file into memory.
+
+    This generator yields log lines one by one, handling both single files
+    and directories with multiple log files.
+
+    Args:
+        zen_store: The store in which the artifact is stored.
+        artifact_store_id: The ID of the artifact store.
+        logs_uri: The URI of the log file or directory.
+
+    Yields:
+        Individual log lines as strings.
+
+    Raises:
+        DoesNotExistException: If the artifact does not exist in the artifact store.
+    """
+    artifact_store = load_artifact_store(artifact_store_id, zen_store)
+
+    try:
+        if not artifact_store.isdir(logs_uri):
+            # Single file case
+            with artifact_store.open(logs_uri, "r") as file:
+                for line in file:
+                    yield line.rstrip("\n\r")
+        else:
+            # Directory case - may contain multiple log files
+            files = artifact_store.listdir(logs_uri)
+            if not files:
+                raise DoesNotExistException(
+                    f"Folder '{logs_uri}' is empty in artifact store "
+                    f"'{artifact_store.name}'."
+                )
```

Comment:
does this really support live-watching the logs while the pipeline is still running, in the context in which multiple log files are created and merged only at the end of the step/pipeline run ? I think I've seen some inconsistencies like older client logs not showing up until the pipeline run is complete but recent logs are available. we have to retest this.